### PR TITLE
User profile#1

### PR DESCRIPTION
**Pull request by AvrAlexandra**

### CODE REVIEW

File: src/main/kotlin/ro/hibyte/betting/service/UserProfileService.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        userProfileRepository.save(userProfile)
+        return userProfile.profilePicture
+    }
+
+    fun getId(userProfile: UserProfile): Long?{
+        return userProfile.userId}
+
+    fun createUserProfileIfNonExistent(userProfileDTO: UserProfileDTO): UserProfile{
+        val userId: Long = userProfileDTO.userId?:0
+        var userProfile = userProfileRepository.findById(userId)
```
Comment (DrBaxR) [inline src/main/kotlin/ro/hibyte/betting/service/UserProfileService.kt:57]: Here I would get rid of the `if` and do something like `return userProfileRepository.findById(userId).orElse(/*...*/)`

File: src/main/kotlin/ro/hibyte/betting/service/WaspService.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+@Service
+class WaspService {
+
+    private val restTemplate = RestTemplate()
+
+    fun sendPhotoToWasp(photo: MultipartFile): Long {
+
+        val headers = HttpHeaders().apply {
+            contentType = MediaType.MULTIPART_FORM_DATA
+            set("X-API-KEY", "a4c1098d-9fda-4b27-b268-62a066f0cddd")
```
Comment (DrBaxR) [inline src/main/kotlin/ro/hibyte/betting/service/WaspService.kt:25]: Extract to configuration (let me know when you get to this and I'll give you more details)

File: src/main/kotlin/ro/hibyte/betting/service/WaspService.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            override fun getFilename(): String {
+                return photo.originalFilename ?: "photo"
+            }
+        }
+        val body: MultiValueMap<String, Any> = LinkedMultiValueMap<String, Any>().apply {
+            add("files", fileResource)
+        }
+        val requestEntity: HttpEntity<MultiValueMap<String, Any>> = HttpEntity(body, headers)
+        val response = restTemplate.exchange(
+            "https://wasp.live.hibyte.ro/api/files",
```
Comment (DrBaxR) [inline src/main/kotlin/ro/hibyte/betting/service/WaspService.kt:38]: Extract to config

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -2,7 +2,7 @@ spring:
   datasource:
     url: jdbc:postgresql://localhost:5432/betting
     username: postgres
-    password: admin
+    password: 210902
```
Comment (DrBaxR) [inline src/main/resources/application.yml:5]: Revert this to initial value and try to not commit your passwords (let me know when you get to this and I'll show you a trick that helps with this)

**Review by DrBaxR — CHANGES_REQUESTED** [review]
Ignore the comments that I left on files that are not yours/that don't apply to your files.

**Review by DrBaxR — APPROVED** [review]

**Review by ambragoanta — APPROVED** [review]

**Pull request merged**